### PR TITLE
REFACTOR: remove the shell commands

### PR DESCRIPTION
**Pull request by amtoine**
Related to #8368.

# Description
as planned in #8311, the `enter`, `shells`, `g`, `n` and `p` commands have been re-implemented in pure-`nushell` in the standard library.
this PR removes the `rust` implementations of these commands.

- all the "shells" tests have been removed from `crates/nu-commnand/tests/commands/` in 2cc6a82da6f77c01f2a0d06847eabde168a2e56b, except for the `exit` command
- `cd` does not use the `shells` feature in its source code anymore => that does not change its single-shell behaviour
- all the command implementations have been removed from `crates/nu-command/src/shells/`, except for `exit.rs` => `mod.rs` has been modified accordingly
- the `exit` command now does not compute any "shell" related things
- the `--now` option has been removed from `exit`, as it does not serve any purpose without sub-shells

# User-Facing Changes
users may now not use `enter`, `shells`, `g`, `n` and `p`
now they would have to use the standard library to have access to equivalent features, thanks to the `dirs.nu` module introduced by @bobhy in #8368 

# Tests + Formatting
- :green_circle: `toolkit fmt`
- :green_circle: `toolkit clippy`
- :black_circle: `toolkit test`
- :black_circle: `toolkit test stdlib`

# After Submitting
the website will have to be regenerated to reflect the removed commands :+1:

### CODE REVIEW

**Comment by amtoine** [comment]
@kubouch
as in #8406, maybe this PR should be marked with the `std-library` tag as well? :yum:

**Comment by fdncred** [comment]
It may be good to look at the tests here and make sure we have similar tests on the stdlib side.

**Comment by amtoine** [comment]
> It may be good to look at the tests here and make sure we have similar tests on the stdlib side.
>
:100:

i can file an issue to list these tests?

**Comment by bobhy** [comment]
Can we hold off on merging this until we have an easy way for user to
install the standard library?  I'm working on a manual install that should
be ready real soon now, then we could document a fairly smooth transition
for users.

**Comment by amtoine** [comment]
> Can we hold off on merging this until we have an easy way for user to install the standard library? I'm working on a manual install that should be ready real soon now, then we could document a fairly smooth transition for users.
>
do not worry, this will stay a DRAFT until that time has come :wink: 

and it's mentionned at the very top in the ":warning: **Warning**" :yum:

**Comment by fdncred** [comment]
I have a tiny bit of hesitancy because we're essentially replacing the shells commands with nearly identical clones but we're changing all the names. I'm wondering if we should either name the dirs commands to match or make aliases to help shells users get up to speed without spending time figuring out what is broken? Interested in other's thoughts.

**Comment by bobhy** [comment]
Hmm, "Welcome to the wonderful world of pre V1 software, where there
are no promises of backward compatibility?"
I think a release note documenting the lossage and the replacement
aliases should suffice.

**Comment by amtoine** [comment]
i do not have a strong preference :thinking: :yum: 

- i feel the "shell" commands are primarily used in an interactive REPL, so people that use them would only have to change muscle memory / use custom aliases, unlikely fix scripts
- we can also fallback to the "legacy" names / ship "standard" aliases for backwards compatibility

**Comment by amtoine** [comment]
what do we do here in the end @fdncred?
i do not use these commands, so i'm fine with either way :innocent:

**Comment by fdncred** [comment]
i'd vote to land it after the next release in a coupld days if we can get synonym aliases for defs for backwards compatibility for `enter`, `p`, `n`, `g`. this probably also needs a `git merge main` since it's a bit old now.

**Comment by amtoine** [comment]
> i'd vote to land it after the next release in a coupld days
>
let's go for this then :yum: 

> if we can get synonym aliases for defs for backwards compatibility for `enter`, `p`, `n`, `g`.
>
do you mean adding Nushell aliases in `default_config.nu`? :thinking: 

> this probably also needs a `git merge main` since it's a bit old now.
>
there is no conflict, you just want to re run the CI with the latest `main` for integration?

**Comment by fdncred** [comment]
> do you mean adding Nushell aliases in default_config.nu? 🤔

I meant in the module itself so that anyone who uses it will have the backwards compatible usage.

> there is no conflict, you just want to re run the CI with the latest main for integration?

Even though the CI is green, there have been changes since the PR as submitted and sometimes we get into trouble if we have PRs that have set for a long time and then land them.

**Comment by kubouch** [comment]
I think the `dirs` module could export the `n`, `p` etc. as aliases to the `dirs` commands and these could be added to the prelude to be loaded automatically. This way we don't rely on users periodically checking and updating their config after every release.

**Comment by fdncred** [comment]
> I think the dirs module could export the n, p etc. as aliases to the dirs commands and these could be added to the prelude to be loaded automatically. This way we don't rely on users periodically checking and updating their config after every release.

This is what I meant, just said better. 😄

**Comment by amtoine** [comment]
first, let's run the CI... :recycle:

**Comment by amtoine** [comment]
this should be better with the last two commits :relieved:

**Comment by fdncred** [comment]
@amtoine I think this is close.

`g` isn't `dirs show`. `g` goes to a particular index. It looks like maybe we need a `dirs goto` something like:
```
alias g = dirs goto
def 'dirs goto' [idx: int] {
  # bounds checking is done in _fetch
  _fetch $idx
}
```

or maybe just
```
alias g = _fetch
```

I'm not sure either of these work but I think you get my point.

We have a `dirs drop` which works like `exit` with the shells commands but I don't think we should alias `exit` to something else. So, this is kind of a missing link. maybe `alias dexit = dirs drop`?

**Comment by amtoine** [comment]
very good points @fdncred 
- i think `g` should be closer to the old built-in one now with 662056e9b
- and new `dexit` alias in f21a64bb1

File: crates/nu-std/lib/dirs.nu [hunk 1]
```diff
@@ -74,7 +76,29 @@ export def-env "dirs show" [] {
...
+            label: {
+                text: $"`shell` should be between 0 and (($env.DIRS_LIST | length) - 1)"
+                start: $span.start
+                end: $span.end
+            }
+        }
+    }
+    let-env DIRS_POSITION = $shell
+
+    cd ($env.DIRS_LIST | get $env.DIRS_POSITION)
```
Comment (fdncred) [inline crates/nu-std/lib/dirs.nu:98]: I'm not sure if all this is necessary when we already have this type of checking in `_fetch`, just done differently. Am I missing something? I do like checking for null and doing `dirs show` if it's null.

Comment (amtoine) [reply crates/nu-std/lib/dirs.nu:98]: it's not really the same
- `_fetch` changes the current shell by an offset, e.g. by `+2` or `-3` and then wrapping around with `mod` the size of the current list of shells
- built-in `g` does not go by relative offset but rather by absolute index, hence the new `dirs goto` which does this exactly

:bulb: as stated in the help of built-in `g`
> Switch to a given shell, or list all shells if no given shell number.
>
and
> shell_number <one_of(int, string)>: shell number to change to (optional)
>
whereas `_fetch` uses an `offset: int`

does that answer your concern? :blush:

Comment (fdncred) [reply crates/nu-std/lib/dirs.nu:98]: Yes, it does. Thanks for explaining it.

**Comment by fdncred** [comment]
ok, let's try this removal and see if anyone notices.

File: crates/nu-std/lib/dirs.nu (hunk 1 above)
Comment (amtoine) [reply crates/nu-std/lib/dirs.nu:98]: :relieved:

**Comment by fdncred** [comment]
> because we added the compatibility aliases, i'm wondering if this still is a breaking change 🤔

i think so, if only because exit is not dexit for the new shells commands. or maybe mostly-non-breaking, and only 1 command is a breaking change. not sure of a good way to state that.

**Comment by amtoine** [comment]
if there is a single tiny detail which is a breaking change, then the whole thing is a breaking change :relieved:

**Pull request merged**